packages/presets/graphql-modules/tests/integration.spec.ts

```diff
@@ -199,10 +199,35 @@ describe('Integration', () => {
     `);
   });
 
-  test('import paths for ESM should have correct extension', async () => {
+  test('import paths for ESM - does not add extension by default', async () => {
+    const output = await executeCodegen(options);
+    const esmImportStatement = `import * as Types from "../global-types";`;
+
+    expect(output.length).toBe(5);
+    expect(output[1].content).toMatch(esmImportStatement);
+    expect(output[2].content).toMatch(esmImportStatement);
+    expect(output[3].content).toMatch(esmImportStatement);
+    expect(output[4].content).toMatch(esmImportStatement);
+  });
+
+  test('import paths for ESM - does not add extension if emitLegacyCommonJSImports=true', async () => {
+    const output = await executeCodegen({
+      ...options,
+      emitLegacyCommonJSImports: true,
+    });
+    const esmImportStatement = `import * as Types from "../global-types";`;
+
+    expect(output.length).toBe(5);
+    expect(output[1].content).toMatch(esmImportStatement);
+    expect(output[2].content).toMatch(esmImportStatement);
+    expect(output[3].content).toMatch(esmImportStatement);
+    expect(output[4].content).toMatch(esmImportStatement);
+  });
+
+  test('import paths for ESM - adds .js extension if emitLegacyCommonJSImports=false', async () => {
     const emitLegacyCommonJSImports = {
-      emitLegacyCommonJSImports: false,
       ...options,
+      emitLegacyCommonJSImports: false,
```

Comment:
`emitLegacyCommonJSImports` should be added after the spread to avoid someone accidentally setting `emitLegacyCommonJSImports` in `options` and overriding the test setup